### ISSUE TITLE
Spec [ postgresql-apb ] failed validation for the following reason: [ APB Spec version [1.0.0] out of bounds 1.0 <= 1.0 ]

### ISSUE THREAD

**Issue by jmontleon** [fusor/ansible-playbook-bundle#320]
With the latest image of ASB / OAB  Broker deployed on origin 3.10 using command "oc cluster add automation-service-broker", we can't use anymore some services such as `postgresql-apb` as there is a spec validation error

```
time="2018-09-11T09:14:08Z" level=warning
msg="Spec [ postgresql-apb ] failed validation for the following reason: 
[ APB Spec version [1.0.0] out of bounds 1.0 <= 1.0 ]. 
It will not be made available."
```

Broker log : https://gist.github.com/cmoulliard/789b36c015f927f66a8ec4d1c58400d0#file-gistfile1-txt-L379

**Comment by jmontleon** [comment]
@cmoulliard

**Comment by jmontleon** [comment]
The APB specs for these APB's were updated to go along with:
https://github.com/automationbroker/bundle-lib/commit/78e6a9409289cc32623e792f0ff0d9eff48a9ba8

We either need to set them to 1.0 to be reverse compatible or bring this change into the release-1.2/3.10 broker to fix this. It does not look like bundle-lib was forked for 1.2 from what I can see, unless this was the intention of the release-0.1 branch?

@dzager @shawn-hurley @eriknelson @jmrodi How would you like to handle this? It might be easiest to set the version on the APB's to 1.0 if there's no reason to bump the version beyond 1.0/1.0.0?

**Comment by cmoulliard** [comment]
This is really blocking for us as we can't continue to develop our devtools, guides, .... Is there a workaround when we use `oc cluster up` and origin `3.10` ?

**Comment by geoand** [comment]
A workaround would be greatly appreciated :)

**Comment by jmontleon** [comment]
I created this on the wrong repo myself; sorry should have been openshift/ansible-service-broker. Haven't had enough coffee yet. I'll leave it here for now so we can walk through it

**Comment by leifmadsen** [comment]
I've seen this same thing on a handful of APBs being importing into OpenShift 3.9 as well.

```
[2018-09-11T15:03:06.28Z] [INFO] - Validating specs...
[2018-09-11T15:03:06.281Z] [WARNING] - Spec [ mysql-apb ] failed validation for the following reason: [ APB Spec version [1.0.0] out of bounds 1.0 <= 1.0 ]. It will not be made available.
[2018-09-11T15:03:06.281Z] [WARNING] - Spec [ postgresql-apb ] failed validation for the following reason: [ APB Spec version [1.0.0] out of bounds 1.0 <= 1.0 ]. It will not be made available.
[2018-09-11T15:03:06.281Z] [WARNING] - Spec [ mediawiki-apb ] failed validation for the following reason: [ APB Spec version [1.0.0] out of bounds 1.0 <= 1.0 ]. It will not be made available.
[2018-09-11T15:03:06.281Z] [WARNING] - Spec [ mariadb-apb ] failed validation for the following reason: [ APB Spec version [1.0.0] out of bounds 1.0 <= 1.0 ]. It will not be made available.
[2018-09-11T15:03:06.281Z] [WARNING] - 4 specs of 43 discovered specs failed validation from registry: docker.io
```

**Comment by Asgoret** [comment]
yeah.... OKD 3.11 same thing... is there any help from RH or developers of this broker? everywhere is old guides or examples...and in ansible-galaxy 60%+ of examples are broken(((

**Comment by djzager** [comment]
> yeah.... OKD 3.11 same thing... is there any help from RH or developers of this broker? everywhere is old guides or examples...and in ansible-galaxy 60%+ of examples are broken(((

I'm sorry that this is causing you trouble. It would help me if you expanded on this "everywhere is old guides or examples". The examples, particular postgresql-apb, mariadb-apb, mediawiki-apb, and mysql-apb have been updated and these _should_ also be what you see in ansible-galaxy. The specific problem here is that older brokers (v3.9 and v3.10) are being pointed at new APBs (latest). I suspect that even in the case of OKD 3.11 if you are using the automation-broker component you are getting the `v3.10` version of the broker that is not compatible with `latest` APBs.

I'm curious, in this case, how you are deploying the broker? Is it via `oc`?

**Comment by Asgoret** [comment]
> > yeah.... OKD 3.11 same thing... is there any help from RH or developers of this broker? everywhere is old guides or examples...and in ansible-galaxy 60%+ of examples are broken(((
> 
> I'm sorry that this is causing you trouble. It would help me if you expanded on this "everywhere is old guides or examples". The examples, particular postgresql-apb, mariadb-apb, mediawiki-apb, and mysql-apb have been updated and these _should_ also be what you see in ansible-galaxy. The specific problem here is that older brokers (v3.9 and v3.10) are being pointed at new APBs (latest). I suspect that even in the case of OKD 3.11 if you are using the automation-broker component you are getting the `v3.10` version of the broker that is not compatible with `latest` APBs.
> 
> I'm curious, in this case, how you are deploying the broker? Is it via `oc`?
Hi! Thanks for quick response.

In my keys i have folder `dev` and APB scan it for version. When i delete it it works fine) At this moment i seach information how to write more complicated playbooks with different plans and environment.

 I use minishift, so the APB component is latest (i think). When i talk about documentation i mean OKD docks. They still start from `apb init` and there is no links to github with new documentation. Also when you instal APB component you get a lot of APB examples (like prometheus and etc) and they didn't work. I mean DB is simple examples  and something with GUI, different plans is more instesresting for learning and this examples doesn't work...
